main/src/ca/uwaterloo/flix/language/phase/unification/set/SetFormula.scala

```diff
@@ -406,116 +385,11 @@ object SetFormula {
-
-    var workList = fs
-    while (workList.nonEmpty) {
-      val f0 :: next = workList
-      workList = next
-      f0 match {
-        case Univ =>
-          ()
-        case Empty =>
-          return Empty
-        case c@Cst(_) =>
-          if (cstsNeg.contains(c)) return Empty
-          cstsPos += c
-        case Compl(c@Cst(_)) =>
-          if (cstsPos.contains(c)) return Empty
-          cstsNeg += c
-        case ElemSet(s) =>
-          elemPos0 = CofiniteIntSet.intersection(elemPos0, s)
-          if (elemPos0.isEmpty) return Empty
-        case Compl(ElemSet(s)) =>
-          elemPos0 = CofiniteIntSet.difference(elemPos0, s)
-          if (elemPos0.isEmpty) return Empty
-        case x@Var(_) =>
-          if (varsNeg.contains(x)) return Empty
-          varsPos += x
-        case Compl(x@Var(_)) =>
-          if (varsPos.contains(x)) return Empty
-          varsNeg += x
-        case Inter(elemPos1, cstsPos1, varsPos1, elemNeg1, cstsNeg1, varsNeg1, other1) =>
-          // To avoid wrapping negated subformulas, we process them inline
-          for (e <- elemNeg1) {
-            elemPos0 = CofiniteIntSet.difference(elemPos0, e.s)
-            if (elemPos0.isEmpty) return Empty
-          }
-          for (x <- cstsNeg1) {
-            if (cstsPos.contains(x)) return Empty
-            cstsNeg += x
-          }
-          for (x <- varsNeg1) {
-            if (varsPos.contains(x)) return Empty
-            varsNeg += x
-          }
-          // Add the positive subformulas to the work list
-          workList = elemPos1.toList ++ cstsPos1 ++ varsPos1 ++ other1 ++ workList
-        case union@Union(_) =>
-          other += union
-        case xor@Xor(_) =>
-          other += xor
-        case compl@Compl(_) =>
-          other += compl
-      }
-    }
-    // Split into pos/neg elements.
-    val (elemPos, elemNeg) = elemPos0 match {
-      case CofiniteIntSet.Set(s) =>
-        if (s.isEmpty) {
-          // empty ∩ ..
-          // = empty       (idempotent intersection formula)
-          // Could return Empty, but unreachable since we check for empty at each operation.
-          throw InternalCompilerException(s"Impossible empty ElemSet $s", SourceLocation.Unknown)
-        } else {
-          // s.nonEmpty
-          // s ∩ ..
-          (Some(ElemSet(s)), None)
-        }
-      case CofiniteIntSet.Compl(s) =>
-        if (s.isEmpty) {
-          // !empty ∩ ..
-          // = univ ∩ ..   (complement distribution)
-          // = ..          (neutral intersection formula)
-          (None, None)
-        } else {
-          // s.nonEmpty
-          // !s ∩ ..
-          (None, Some(ElemSet(s)))
-        }
-    }
-
-    // Avoid intersections with zero or one subformula.
-    subformulasOf(elemPos, cstsPos, varsPos, elemNeg, cstsNeg, varsNeg, other).take(2).toList match {
-      case Nil => return Univ
-      case one :: Nil => return one
-      case _ => ()
+    fs match {
+      case Nil => Empty
```

Comment:
I guess this case never happens but I think it should be Univ since its the neutral set for intersection

Reply:
I am still programming